### PR TITLE
Place API Implementation

### LINKED ISSUE #34: Implement /place API

### PR DESCRIPTION
**Pull request by msmollin**
Fixes #34

This has a bunch of additional refactoring and new classes for streamlining
along with reducing the number of dependencies required for sample app and extension
pods

Coming in a future push:
- Get the sample app to do more than just print out the response from pelias when querying for a place.
- Look into more/better unit tests of /place - what's there now covers the major functionality but I'm sure I'm missing edge cases

### CODE REVIEW

File: SampleApp/SecondViewController.swift [hunk 1]
```diff
@@ -108,5 +109,30 @@ class SecondViewController: UIViewController, UITextFieldDelegate, CLLocationMan
   func locationManager(manager: CLLocationManager, didFailWithError error: NSError) {
     print(error)
   }
+  
+  //Mapview Delegate
```
Comment (sarahcodes100) [inline SampleApp/SecondViewController.swift]: nit: "// MARK: - Mapview Delegate" might be more useful since you can use it in the class overview view

File: SampleApp/SecondViewController.swift [hunk 2]
```diff
@@ -108,5 +109,30 @@ class SecondViewController: UIViewController, UITextFieldDelegate, CLLocationMan
...
+  func mapView(mapView: MKMapView, viewForAnnotation annotation: MKAnnotation) -> MKAnnotationView? {
+    if annotation.isKindOfClass(MKUserLocation) {
```
Comment (sarahcodes100) [inline SampleApp/SecondViewController.swift:117]: will annotation either descend from MKUserLocation or MKLocation?

File: SampleApp/SecondViewController.swift [hunk 3]
```diff
@@ -108,5 +109,30 @@ class SecondViewController: UIViewController, UITextFieldDelegate, CLLocationMan
...
+      return nil
+    }
+    
+    let annotationView = MKPinAnnotationView(annotation: annotation, reuseIdentifier: "pin_loc")
```
Comment (sarahcodes100) [inline SampleApp/SecondViewController.swift]: nit: string constant for "pin_loc" ?

File: extras/convenience/MapkitExtensions.swift [hunk 4]
```diff
@@ -58,18 +57,46 @@ Example:
...
+    var mapItems = [PeliasMapkitAnnotation]()
     if let jsonDictionary = parsedResponse?.parsedResponse {
       let featuresArray = jsonDictionary["features"] as! [[String:AnyObject]]
       for feature in featuresArray {
         //Address Dictionary for Placemark Creation
         let featureProperties = feature["properties"] as! [String:AnyObject]
         var addressDictionary = [String:String]()
-        addressDictionary[CNPostalAddressCountryKey] = featureProperties["country"] as? String
-        
+        addressDictionary[PeliasIDKey] = featureProperties["id"] as? String
```
Comment (sarahcodes100) [inline extras/convenience/MapkitExtensions.swift:98]: why not have string constants for these keys?

File: src/PeliasPlaceConfig.swift [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+public struct PeliasPlaceConfig : PlaceAPIConfigData {
+  
+  var urlEndpoint = NSURL.init(string: "/v1/place", relativeToURL: PeliasSearchManager.sharedInstance.baseUrl)!
+  var queryItems = [String:NSURLQueryItem]()
+  var completionHandler: (PeliasResponse) -> Void
+  
+  var apiKey: String? {
+    didSet {
+      if let key = apiKey where apiKey?.isEmpty == false {
+        appendQueryItem("api_key", value: key)
```
Comment (sarahcodes100) [inline src/PeliasPlaceConfig.swift:20]: use string constant here too?

File: src/PeliasPlaceConfig.swift [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+  var placeId: String
+  var dataSource: SearchSource
+  var layer: LayerFilter
+  
+  init(placeId: String, dataSource: SearchSource, layer: LayerFilter) {
+    self.placeId = placeId
+    self.dataSource = dataSource
+    self.layer = layer
+  }
+}
```
Comment (sarahcodes100) [inline src/PeliasPlaceConfig.swift]: i just had this issue in a recent pr- looks like you need a trailing new line at the end of the file

File: SampleApp/SecondViewController.swift (hunk 2 above)
Comment (msmollin) [inline SampleApp/SecondViewController.swift:117]: Indeed. When user location is enabled on a map, the blue pulsing "pin" is also an annotation, and so for most purposes you want to filter that out.

File: SampleApp/SecondViewController.swift (hunk 3 above)
Comment (msmollin) [inline SampleApp/SecondViewController.swift]: Certainly, although it's not used anywhere else yet. Fixed.

File: SampleApp/SecondViewController.swift (hunk 1 above)
Comment (msmollin) [inline SampleApp/SecondViewController.swift]: Good idea. Added it for the other line in this file.

File: extras/convenience/MapkitExtensions.swift (hunk 4 above)
Comment (msmollin) [inline extras/convenience/MapkitExtensions.swift:98]: I do for the "parsed" side of things, as you can see. This particular area needs improvement in general, because basically I'm having to traverse a GeoJSON schema. So the upside here is since Pelias follows the schema, these names won't change much if at all. 

However, what I really wanna do is abstract this out and make use of a real GeoJSON parser, like this one https://github.com/mattt/GeoJSONSerialization

I'm open to thoughts over here on the topic: https://github.com/pelias/pelias-ios-sdk/issues/4

File: src/PeliasPlaceConfig.swift (hunk 5 above)
Comment (msmollin) [inline src/PeliasPlaceConfig.swift:20]: Yeah it's probably about time to add a string constants file I guess. Tagged for 1.0.0 - https://github.com/pelias/pelias-ios-sdk/issues/40

File: src/PeliasPlaceConfig.swift (hunk 6 above)
Comment (msmollin) [inline src/PeliasPlaceConfig.swift]: Added.
